### DISCUSSION TITLE
Reduce the time for CLI startup

### DISCUSSION THREAD

**Discussion by naveen521kk** [ManimCommunity/manim#2795] [Suggestions and Proposals]
## Description of the proposed feature
Currently, when running `manim` executable or module using (`py -m manim`), the whole module is imported which is large and takes a lot of time. I propose to move all the CLI handling stuff (ie `manim.cli`) to a new module called `manim_cli` which would not import manim until the command line arguments are parsed. By doing so, we can reduce the startup time.

## How can the new feature be used?
Reduce the time to load `manim` command. If someone just wanted to see the help this would reduce the time they wait because `manim` wouldn't be imported in that case.

## Additional comments
We would need to move the main module to `src` structure for this to work.
```
src \
    - manim
    - manim_cli
```

**Comment by Freyert** [comment]
@naveen521kk is there any explanation as to why `manim` cli currently imports all the entire library? I would expect rendering a scene to only require the libraries imported in the user's python. Not the entire library.

**Comment by naveen521kk** [comment]
> is there any explanation as to why manim cli currently imports all the entire library?

It's been like that forever. Someone should change it.
